src/preamble.js

```diff
@@ -924,6 +924,11 @@ function getWasmImports() {
 #endif
 #endif
 
+#if hasExportedSymbol('__cpp_exception') && !RELOCATABLE
```

Comment:
Why do we check for not relocatable?

Reply:
Because in relocatable mode symbols like this (along with the wasm table and memory) are not exported by the wasm but created in JS and *imported* by each wasm module.

Reply:
In that case wouldn't `hasExportedSymbol` be false when relocatable?

Reply:
Yes we expect hasExportedSymbol to be false in relocatable mode and true for static builds.  Here are are checking for static builds we check for hasExportedSymbol to be true.

Reply:
But then is
```
A = hasExportedSymbol('__cpp_exception') && !RELOCATABLE
```
not equivalent to
```
B = hasExportedSymbol('__cpp_exception')
```
? Looking at the two cases,

* If relocatable, then hasExportedSymbol is false, `A = false && false => false`, and `B = false`.
* If not relocatable, hasExportedSymbol may be true or false, but `!relocatable` is always true, so it does not cause a difference between `A` and `B`.

So can't we do this?
```suggestion
#if hasExportedSymbol('__cpp_exception')
```

Reply:
Sure you could argue these are redundant.  I'm mostly copying the existing logic for `__indirect_function_table` above.